src/main/java/data/GenericData.java

```diff
@@ -456,4 +459,59 @@ public static JSONObject getRestrictions(String testCode) {
         }
         return actualRestrictions;
     }
```

Comment:
There is a fair bit of duplicate logic in these 2 methods, using a flag or additional parameter should allow you to amalgamate them into 1.

Reply:
done